### ISSUE TITLE
build and release ossmc plugin at end of sprint

### ISSUE THREAD

**Issue by jmazzitelli** [kiali/kiali#6706]
Because the Kiali operator will soon be able to install OSSMC , we need to add the plugin to the list of things built at the end of the sprint (so we want to release the latest plugin with the lastest kiali server and latest kiali operator at the time we build our sprint releases). This means we need to update our GitHub Action workflows so the the plugin is built along with the server and operator. For details on the releasing kiali, see [RELEASING.md](https://github.com/kiali/kiali/blob/master/RELEASING.md)

It should work the same way we build the server image.  Build the multi-arch image and push the image to quay. Ideally, the image tag will be the same version as the kiali server and operator (e.g. v1.78.0). If the plugin version numbers will NOT track the version of the operator, we will need to do some custom work in the operator because currently the operator default is to install the same image version as the operator's own version.

see https://issues.redhat.com/browse/OSSM-4944

PRs:
* kiali/kiali: https://github.com/kiali/kiali/pull/6735 (just updates the RELEASING.md document)
* https://github.com/kiali/openshift-servicemesh-plugin/pull/213

**Comment by jmazzitelli** [comment]
Here is the plugin release workflow - https://github.com/kiali/openshift-servicemesh-plugin/blob/main/.github/workflows/release.yaml

Note that once the operator is removed from the plugin repo (see https://github.com/kiali/openshift-servicemesh-plugin/pull/211) the only thing that release workflow will be doing is building the plugin image and pushing the plugin image to quay.

**Comment by jshaughn** [comment]
Is there a way for the console/plugin user to know which version of the plugin they are using, along with the version of Kiali etc?  Basically the ABout box for the plugin?  Also wondering if the plugin should be versioned the same as Kiali, and not independently.  The plugin is basically alternate "ui"and should match the server version.

**Comment by jmazzitelli** [comment]
`kubectl get ossmconsole <name> -n <namespace> -o yaml`

The `status` field will show the version of Kiali that the plugin detected. The `app.kubernetes.io/version` label tells you the plugin version (the version of the image technically).

**Comment by jmazzitelli** [comment]
> wondering if the plugin should be versioned the same as Kiali, and not independently. The plugin is basically alternate "ui"and should match the server version.

In my PRs, I actually do that. I moved the plugin version to match that of the kiali version - see the Makefile change in the PR https://github.com/kiali/openshift-servicemesh-plugin/pull/211/files where it now has:
```make
# Identifies the current build. Match the same version of Kiali Server and Operator.
VERSION ?= v1.76.0-SNAPSHOT
```

> Is there a way for the console/plugin user to know which version of the plugin they are using, along with the version of Kiali etc? Basically the ABout box for the plugin?

This is what the OpenShift Console itself provides. You first go to the OpenShift lefthand nav and select "Administration>Cluster Settings". Then on the right side select "Configuration" tab. From there select the item "Console (operator.openshift.io"). On the resulting page, select the "Console plugins" tag and you will see the version info of the plugin itself)  :

![image](https://github.com/kiali/kiali/assets/2029470/14d17342-aa60-4e15-9863-ef15ca98bbfe)

You can obtain the version of the plugin using the "oc" CLI and getting the version label off the plugin pod. You do this via:

```bash
NAMESPACE="$(oc get ossmconsole --all-namespaces -o jsonpath={..metadata.namespace})"
PODNAME="$(oc get pods -n $NAMESPACE -l app.kubernetes.io/name=ossmconsole -o name)"
oc get $PODNAME -n $NAMESPACE -o jsonpath='{.metadata.labels.app\.kubernetes\.io/version}{"\n"}'
```

To get the version of the Kiali Server that the plugin is using, you can see that information in the status field of the OSSMConsole CR - specifically the `status.environment.kialiVersion` field. You can get this from the "oc" CLI running this command:

`oc get ossmconsole --all-namespaces -o jsonpath='{..status.environment.kialiVersion}{"\n"}'`

The result will be the actual Kiali Server version:

```
$ oc get ossmconsole --all-namespaces -o jsonpath='{..status.environment.kialiVersion}{"\n"}'
v1.76.0-SNAPSHOT
```

So, in short, we could provide this small script (or at least document these commands) to get the plugin version and the kiali version (for the kiali that the plugin is using):

```bash
NAME="$(oc get ossmconsole --all-namespaces -o jsonpath={..metadata.name})"
NAMESPACE="$(oc get ossmconsole --all-namespaces -o jsonpath={..metadata.namespace})"
PODNAME="$(oc get pods -n $NAMESPACE -l app.kubernetes.io/name=ossmconsole -o name)"
echo "Plugin version =" $(oc get $PODNAME -n $NAMESPACE -o jsonpath='{.metadata.labels.app\.kubernetes\.io/version}{"\n"}')
echo "Kiali version =" $(oc get ossmconsole --all-namespaces -o jsonpath='{..status.environment.kialiVersion}{"\n"}')
```
